src/nvim/option.c

```diff
@@ -2578,10 +2578,32 @@ did_set_string_option (
     if (check_opt_strings(p_ead, p_ead_values, FALSE) != OK)
       errmsg = e_invarg;
   }
+  // 'clipboard'
   else if (varp == &p_cb) {
-    if (opt_strings_flags(p_cb, p_cb_values, &cb_flags, TRUE) != OK)
+    if (opt_strings_flags(p_cb, p_cb_values, &cb_flags, TRUE) != OK) {
       errmsg = e_invarg;
-  } else if (varp == &(curwin->w_s->b_p_spl)
+    }
+    if (errmsg == NULL) {
```

Comment:
I didn't see `clip_unnamed` was replaced with `cb_flags`, so I tried to restore it here. I'll remove this.